app/main.py

```diff
@@ -1,3 +1,20 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+from functools import wraps
+
+
+def cache(func: Callable) -> Callable:
+    cache_hash_result = {}
+
+    @wraps(func)
+    def results_save_wrapper(*args, **kwargs) -> None:
```

Comment:
I don't agree that the function returns None )

Reply:
![image](https://user-images.githubusercontent.com/107141441/216668439-b759477f-ce76-4bb9-9170-aa45cf91dab1.png)
Any? Union[Any | None] ? not sure which is better)

Reply:
`Any` will be enough )